src/service/vaults.ts

```diff
@@ -92,6 +91,78 @@ const watchPriceFeeds = () => {
   };
 };
 
+type VaultSubscribers = {
+  asset: string;
+  vault: string;
+};
+
+const watchUserVaults = () => {
+  let isStopped = false;
+  const watchedVaults = new Set<string>();
+
+  const watchVault = async (
+    offerId: string,
+    subscriber: string,
+    managerId: string,
+  ) => {
+    useVaultStore.getState().markVaultForLoading(offerId, managerId);
+
+    const { leader, importContext } = appStore.getState();
+    const f = makeFollower(`:${subscriber}`, leader, {
+      unserializer: importContext.fromBoard,
+    });
+
+    for await (const { value } of iterateLatest<VaultUpdate>(f)) {
+      if (isStopped) break;
+      console.debug('got update', subscriber, value);
+      useVaultStore
+        .getState()
+        .setVault(offerId, { ...value, managerId, isLoading: false });
+    }
+  };
+
+  const watchNewVaults = async (
+    subscribers: Record<string, Record<string, string>>,
+  ) => {
+    if (!useVaultStore.getState().vaults) {
+      useVaultStore.setState({ vaults: new Map() });
+    }
+
+    Object.entries(subscribers).forEach(([offerId, subscribers]) => {
+      // XXX: If a third party contract returns a similar looking offer result,
+      // it could trick the UI into thinking the user has a vault. Is there a
+      // better way to filter for real vaults the user owns?
+      const vaultSubscriber = (subscribers as VaultSubscribers).vault;
+      if (!vaultSubscriber || watchedVaults.has(offerId)) {
+        return;
+      }
+
+      watchedVaults.add(offerId);
+      const managerId = (subscribers as VaultSubscribers).asset
+        .split('.')
+        .pop();
```

Comment:
this string logic should be split out into a function to make obvious what it's doing. it might also change before release and it would be helpful to have isolated.

Reply:
Done